Werewolf for Telegram/Werewolf Control/Commands/AdminCommands.cs

```diff
@@ -23,6 +23,114 @@ namespace Werewolf_Control
 {
     public static partial class Commands
     {
+        [Attributes.Command(Trigger = "setgrouptopic", GroupAdminOnly = true, InGroupOnly = true)]
+        public static void SetGroupTopic(Update update, string[] args)
+        {
+            long chatId = update.Message.Chat.Id;
+            int? topicId = update.Message.MessageThreadId;
+
+            int? currentTopicId;
+
+            using (var db = new WWContext())
+            {
+                currentTopicId = db.Groups
+                    .Where(g => g.GroupId == chatId)
+                    .Select(g => g.GroupTopicId)
+                    .FirstOrDefault();
+            }
+
+            bool inGeneralTopic = topicId == null || topicId == 0;
+
+            string infoText = $"Current Topic ID: `{(currentTopicId.HasValue ? currentTopicId.Value.ToString() : "None")}`\n\n";
+
+            if (inGeneralTopic)
+            {
+                infoText += "This message is in the general topic (no thread ID).\n" +
+                            "You cannot set this as the group topic. Only specific threads can be set.";
+            }
+            else
+            {
+                infoText += $"> You are currently inside topic ID: `{topicId}`.\n" +
+                            "Choose what you'd like to do:";
+            }
```

Comment:
Please work with localized strings here. You can get `English.xml` by sending `/getlang English` to the bot, and add the strings with placeholders as required accordingly. The changed `English.xml` need not be part of the pull request, but you can provide it to us for uploading it after you have made the changes.

Reply:
i will do this after every other things are resolved